warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/job/ShardedTableMapFileTest.java

```diff
@@ -188,5 +226,109 @@ public void testWriteSplitsFileExistingPathMultipleKeyExtents() throws Exception
         Assert.assertEquals("location2_1234", val.toString());
         valid = reader.next(key, val);
         Assert.assertFalse(valid);
+        reader.close();
+    }
+    
+    @Test
+    public void testSingleDaySplitsCreated_AndValid() throws Exception {
+        String tableName = "validSplits";
+        SortedMap<Text,String> splits = createDistributedLocations(tableName);
+        createSplitsFile(splits, conf, splits.size(), tableName);
+        Map<Text,String> locations = ShardedTableMapFile.getShardIdToLocations(conf, tableName);
+        // three days of splits, all should be good, none of these should error
+        ShardedTableMapFile.validateShardIdLocations(conf, tableName, 0, locations);
+        ShardedTableMapFile.validateShardIdLocations(conf, tableName, 1, locations);
+        ShardedTableMapFile.validateShardIdLocations(conf, tableName, 2, locations);
+    }
+    
+    @Test(expected = IllegalStateException.class)
+    public void testMissingAllOfTodaysSplits() throws Exception {
+        String tableName = "missingTodaysSplits";
+        SortedMap<Text,String> splits = simulateMissingSplitsForDay(0, tableName);
+        createSplitsFile(splits, conf, splits.size(), tableName);
+        Map<Text,String> locations = ShardedTableMapFile.getShardIdToLocations(conf, tableName);
+        // three days of splits, today should be invalid, which makes the rest bad too
+        ShardedTableMapFile.validateShardIdLocations(conf, tableName, 0, locations);
```

Comment:
if ,0, doesn't fail but ,1, does then this test still passes

Reply:
yep, adding a fail() call after each of these